bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieServer.java

```diff
@@ -249,6 +266,12 @@ public void run() {
                 }
                 if (!isBookieRunning()) {
                     shutdown();
+                    if (null != uncaughtExceptionHandler) {
+                        uncaughtExceptionHandler.uncaughtException(
+                            this,
+                            new BookieDeadException("Bookie is not running any more")
+                                .fillInStackTrace());
```

Comment:
Why do we need fillInStackTrace() here? 
Stack trace should be filled in by the Throwable's constructor already, assuming BookieException called super's constructor.

Now this got me looking at BookieException.
"public BookieException(int code)" sets private field code but does not call super().
All other constructors call super(..) variants but don't set code so we cannot rely on getCode(). Luckily, getCode() is not used so why do we even bother passing the code and having the field?

Reply:
> Why do we need fillInStackTrace() here?
Stack trace should be filled in by the Throwable's constructor already, assuming BookieException called super's constructor.

good question. I think some of the constructors are not calling super. I can fix the bookie exception constructors and remove `fillInStackTrace`

> variants but don't set code so we cannot rely on getCode()

we should set the code. will fix it.

> Luckily, getCode() is not used so why do we even bother passing the code and having the field?

the existing of the code is following BKException. will fix the constructors.